test/org/sagebionetworks/bridge/services/AuthenticationServiceMockTest.java

```diff
@@ -261,6 +263,58 @@ public void signIn() {
         verify(accountSecretDao).createSecret(AccountSecretType.REAUTH, USER_ID, REAUTH_TOKEN);
     }
     
+    @Test(expected = ConsentRequiredException.class)
+    public void signInThrowsConsentRequiredException() {
+        study.setReauthenticationEnabled(true);
+        
+        AccountSubstudy as1 = AccountSubstudy.create(TestConstants.TEST_STUDY_IDENTIFIER, "substudyA", USER_ID);
+        AccountSubstudy as2 = AccountSubstudy.create(TestConstants.TEST_STUDY_IDENTIFIER, "substudyB", USER_ID);
+        
+        account.setReauthToken("REAUTH_TOKEN");
+        account.setHealthCode(HEALTH_CODE);
+        account.setAccountSubstudies(ImmutableSet.of(as1, as2));
+        account.setId(USER_ID);
+        
+        CriteriaContext context = new CriteriaContext.Builder()
+            .withStudyIdentifier(TestConstants.TEST_STUDY)
+            .withLanguages(LANGUAGES)
+            .withClientInfo(ClientInfo.fromUserAgentCache("app/13"))
+            .withIpAddress("127.1.1.11").build();
+        
+        doReturn(account).when(accountDao).authenticate(study, EMAIL_PASSWORD_SIGN_IN);
+        doReturn(PARTICIPANT_WITH_ATTRIBUTES).when(participantService).getParticipant(study, account, false);
+        doReturn(UNCONSENTED_STATUS_MAP).when(consentService).getConsentStatuses(contextCaptor.capture(), any());
+        doReturn(REAUTH_TOKEN).when(service).generateReauthToken();
+        doReturn("SESSION_TOKEN").when(service).getGuid();
+        doReturn(Environment.PROD).when(config).getEnvironment();
+        
+        UserSession session = service.signIn(study, context, EMAIL_PASSWORD_SIGN_IN);
```

Comment:
This test almost certainly doesn't do what you expect it to do. If signIn() throws, none of the asserts after this line will be executed.

Reply:
Catching the exception so we can test that session and secret are still handled property.